### ISSUE TITLE
#[inline] is allowed on structs inside fns

### ISSUE THREAD

**Issue by tsion** [rust-lang/rust#31769]
http://is.gd/mRtTeS causes an error, as expected:

```rust
#[inline] // error: attribute should be applied to function
struct Foo(usize);
    
fn main() {
}
```

http://is.gd/POCLQo should cause the same error, but is allowed:

```rust
fn main() {
    #[inline]
    struct Foo(usize);
}
```

**Comment by Aatch** [comment]
Ah, easy fix. [This function](https://github.com/rust-lang/rust/blob/master/src/librustc/front/check_attr.rs#L99-L104) needs to have `visit::walk_item` at the end to walk the contents on the item in question.

**Comment by nxnfufunezn** [comment]
taking this..

**Comment by eddyb** [comment]
(Oops, GitHub UI is finicky. I wanted to check whether I can assign @nxnfufunezn to this issue, but it won't let me)
